pycbc/waveform/compress.py

```diff
@@ -260,24 +258,148 @@ def compress_waveform(htilde, sample_points, tolerance, interpolation,
+    # Converting them into Hz by diving them by total mass of binary in
+    # secs
+    amp_freq_points = amp_freq_points/Mtot_sec
+    phase_freq_points = phase_freq_points/Mtot_sec
+    fmin_interp = max(amp_freq_points.min(), phase_freq_points.min())
+
+    # Perform the frequency space interpolation to get the decompressed waveform
+    hdecomp = fd_decompress(amp_interp_points, phase_interp_points,
+                            phase_freq_points, amp_freq_points, out=None,
+                            df=deltaF, f_lower=fmin_interp,
+                            interpolation=interpolation)
+
+    # Calculate the highest frequency point in the frequency
+    # series for the full decompresssed SEOBNRv2_ROM waveform (htilde)
+    # and the highest frequency point in the frequency series
+    # for the partially decompressed SEOBNRv2_ROM waveform (hdecomp)
+    fmax_hdecomp = numpy.amax(hdecomp.sample_frequencies.numpy())
+    fmax_htilde = numpy.amax(htilde.sample_frequencies.numpy())
+
+    # Calculate the minimum of the highest frequency points for the two
+    # waveforms which would be used as a high_frequency_cutoff while
+    # calculating the mismatch between the two
+    high_frequency_cutoff = min(fmax_hdecomp, fmax_htilde)
+
+    # Reduce htilde and hdecomp to a length that would have the lowest
+    # frequency as the one taken as input in `pycbc_compress_bank` and the
+    # the highest frequency to be the one which is smaller between the
+    # highest frequency taken as input in `pycbc_compress_bank` and the
+    # high_frequency_cutoff calculated above.
+    high_frequency_cutoff_idx = int(high_frequency_cutoff/deltaF)
+    kmin = int(fLow/deltaF)
+    kmax = int(fHigh/deltaF)
+    if high_frequency_cutoff_idx < kmax :
+        kmax = high_frequency_cutoff_idx
+    hdecomp_cut = hdecomp[kmin:kmax]
+    htilde_cut = htilde[kmin:kmax]
+
+    # Calculated the match and then the mismatch between htilde and
+    # hdecomp
+    match, _ = filter.match(hdecomp_cut, htilde_cut, low_frequency_cutoff=fLow, high_frequency_cutoff=high_frequency_cutoff)
+    logging.info("match=%.10f", match)
+    mismatch = 1.0-match
+    logging.info("mismatch: %.10f, for low_frequency_cutoff = %.1f and high_frequency_cutoff = %.1f", \
+                 mismatch, fLow, high_frequency_cutoff)
+    return CompressedWaveform(amp_interp_points, phase_interp_points,
+                              phase_freq_points, amp_freq_points,
+                              interpolation=interpolation,
+                              mismatch=mismatch)
 
+compression_algorithms = {
+        'mchirp': mchirp_compression,
+        'spa': spa_compression,
+        'partial_compress_rom:SEOBNRv2_ROM_DoubleSpin' : partial_rom_compression,
+        'spa:TaylorF2' : spa_compression,
+        'spa:SEOBNRv2_ROM_DoubleSpin' : spa_compression,
+        'mchirp:TaylorF2' : mchirp_compression,
+        'mchirp:SEOBNRv2_ROM_DoubleSpin' : mchirp_compression
+        }
```

Comment:
The compression algorithms should only list the compression algorithm names, not the approximants.